From: Robert Schwebel <r.schwebel@pengutronix.de>
Subject: [patch] add config option for convert tools

Add a configure option to enable the ???-to-usplash tools. When we cross
build for embedded systems, we might want to disable these tools (they
would have to be built for the "build" machine otherwhise).

Signed-off-by: Robert Schwebel <r.schwebel@pengutronix.de>

---
 Makefile.am      |   12 ++++++++++--
 bogl/Makefile.am |   29 ++++++++++++++++++++---------
 configure.ac     |   14 ++++++++++++++
 3 files changed, 44 insertions(+), 11 deletions(-)

Index: b/Makefile.am
===================================================================
--- a/Makefile.am
+++ b/Makefile.am
@@ -41,10 +41,18 @@ usplash_LDADD = \
 usplash_write_SOURCES = \
 	usplash_write.c
 
+if CONVERT_TOOLS
 %.c: %.bdf
-	./bogl/bdftobogl $< > $@
+	./bogl/bdftousplash $< > $@
 
 %.c: %.png
-	./bogl/pngtobogl $< > $@
+	./bogl/pngtousplash $< > $@
+else
+%.c: %.bdf
+	bdftousplash $< > $@
+
+%.c: %.png
+	pngtousplash $< > $@
+endif
 
 SUFFIXES: .png .bdf
Index: b/configure.ac
===================================================================
--- a/configure.ac
+++ b/configure.ac
@@ -70,10 +70,23 @@ AM_CONDITIONAL(SVGA_BACKEND, test "$CONF
 # FIXME:
 AM_CONDITIONAL(NO_ASM, test false)
 
+AC_MSG_CHECKING([whether to enable convert tools])
+AC_ARG_ENABLE(convert-tools,
+	AS_HELP_STRING([--disable-convert-tools], [disable convert tools]),
+	[
+		case "$enableval" in
+			y | yes) CONFIG_TOOLS=yes ;;
+			*) CONFIG_TOOLS=no ;;
+		esac],
+	[CONFIG_TOOLS=yes])
+AC_MSG_RESULT([${CONFIG_TOOLS}])
+AM_CONDITIONAL(CONVERT_TOOLS, test "$CONFIG_TOOLS" = "yes")
+
 #
 # Checks for libraries.
 #
 
+if test "$CONFIG_TOOLS" = "yes"; then
 ##
 # libgd
 ##
@@ -96,6 +109,7 @@ PKG_CHECK_MODULES([libpng],
 )
 AC_SUBST(libpng_CFLAGS)
 AC_SUBST(libpng_LIBS)
+fi
 
 #
 # Checks for header files.
Index: b/bogl/Makefile.am
===================================================================
--- a/bogl/Makefile.am
+++ b/bogl/Makefile.am
@@ -3,7 +3,10 @@ AM_CFLAGS = \
 	-fPIC -I../
 
 lib_LTLIBRARIES = libbogl.la
-bin_PROGRAMS = bdftobogl pngtobogl bterm reduce-font
+bin_PROGRAMS = bterm reduce-font
+if CONVERT_TOOLS
+bin_PROGRAMS += bdftousplash pngtousplash
+endif
 EXTRA_PROGRAMS = bogl-test bowl-boxes
 
 LIBBOGLSOURCES = \
@@ -32,17 +35,17 @@ LIBRSRCSOURCES = \
 	timBI18.c \
 	tux75.c
 
-bdftobogl_SOURCES = \
+bdftousplash_SOURCES = \
 	bdftobogl.c \
 	$(LIBBOGLSOURCES)
 
-pngtobogl_SOURCES = \
+pngtousplash_SOURCES = \
 	pngtobogl.c
 
-pngtobogl_CFLAGS = \
+pngtousplash_CFLAGS = \
 	$(libpng_CFLAGS)
 
-pngtobogl_LDADD = \
+pngtousplash_LDADD = \
 	-lgd -lm \
 	$(libpng_LIBS)
 
@@ -69,11 +72,19 @@ bterm_LDADD = \
 bterm_DEPENDENCIES = \
 	libbogl.la
 
-%.c: %.bdf bdftobogl
-	./bdftobogl $< > $@
+if CONVERT_TOOLS
+%.c: %.bdf bdftousplash
+	./bdftousplash $< > $@
+
+%.c: %.png pngtousplash
+	./pngtousplash $< > $@
+else
+%.c: %.bdf
+	bdftousplash $< > $@
 
-%.c: %.png pngtobogl
-	./pngtobogl $< > $@
+%.c: %.png
+	pngtousplash $< > $@
+endif
 
 bogl_test_SOURCES = \
 	bogl-test.c \
